src/app/src/main/java/br/com/italoromeiro/arctouchtest/presenters/BasePresenter.java

```diff
@@ -0,0 +1,74 @@
+package br.com.italoromeiro.arctouchtest.presenters;
+
+import android.util.Log;
+
+import org.androidannotations.annotations.Background;
+import org.androidannotations.annotations.Bean;
+import org.androidannotations.annotations.EBean;
+import org.androidannotations.annotations.RootContext;
+
+import br.com.italoromeiro.arctouchtest.presenters.actions.NetworkAction;
+import br.com.italoromeiro.arctouchtest.rest.RestController;
+import br.com.italoromeiro.arctouchtest.views.activities.BaseActivity;
+
+/**
+ * Created by italo on 08/08/16.
+ */
+@EBean
+public class BasePresenter {
+    private static final String TAG = BasePresenter.class.getSimpleName();
+
+    @RootContext
+    BaseActivity mActivity;
+
+    @Bean
+    RestController mApi;
+
+    private NetworkAction mRetriable;
```

Comment:
I felt free to use a neologism in English because the idea here is to express that I have a thing expressing an action I can repeat. Actually, I don't know a better word to describe this.